src/server.c

```diff
@@ -1173,20 +1173,24 @@ static void clientWritePauseDuringOOM() {
      * Pause also evictions, since we might reach below maxmemory after lazyfree */
     if ((getMaxmemoryState(NULL,NULL,NULL,NULL) == C_ERR) &&
         (bioPendingJobsOfType(BIO_LAZY_FREE))) {
-        if (!server.stat_current_client_paused_oom_time) {
+        if (!server.is_pause_write_clients_oom) {
             pauseActions(PAUSE_OOM_THROTTLE, LLONG_MAX,
                          PAUSE_ACTION_EVICT | PAUSE_ACTION_CLIENT_DENYOOM);
             
             elapsedStart(&server.stat_current_client_paused_oom_time);
+            server.is_pause_write_clients_oom = 1;
         }
     } else {
-        if (server.stat_current_client_paused_oom_time) {
+        if (server.is_pause_write_clients_oom) {
             /* If not OOM && lazy-free, then unpause clients */
             unpauseActions(PAUSE_OOM_THROTTLE);
-            
-            /* update statistics & reset current_client_paused_oom_time  */
-            server.stat_total_client_paused_oom_time += elapsedUs(server.stat_current_client_paused_oom_time);
-            server.stat_current_client_paused_oom_time = 0;
+            if (server.stat_current_client_paused_oom_time) {
+                /* update statistics & reset current_client_paused_oom_time  */
+                server.stat_total_client_paused_oom_time += elapsedUs(
+                        server.stat_current_client_paused_oom_time);
```

Comment:
i'm sorry for trolling you, but i now realize that resetting stat_current_client_paused_oom_time in RESETSTAT would cause stat_total_client_paused_oom_time to later jump to a huge number.

i suppose these metrics need to be completely decoupled from the `if` that decides if we want to pause, and instead depend on the eventual state (like we do in evict.c and defrag.c).

p.s. we previously didn't have the is_pause_write_clients_oom variable, and relied on the quick exit in pauseActions / unpauseActions when the state doesn't change, maybe we can keep doing that?

Reply:
Don't be sorry. You are doing holy job. I simplified evaluation stat_total_client_paused_oom_time. lmk if you still have any issues. Thanks.